influxdb3_load_generator/src/commands/query.rs

```diff
@@ -1,54 +1,202 @@
-use std::str::Utf8Error;
+use std::{collections::HashMap, sync::Arc, time::Duration};
 
+use anyhow::Context;
+use bytes::Bytes;
+use chrono::Local;
 use clap::Parser;
-use influxdb3_client::Format;
-use secrecy::ExposeSecret;
-use tokio::io;
+use influxdb3_client::Client;
+use serde_json::Value;
+use tokio::time::Instant;
 
-use super::common::InfluxDb3Config;
+use crate::{
+    query_generator::{create_queriers, Format, Querier},
+    report::QueryReporter,
+    specification::QuerierSpec,
+};
 
-#[derive(Debug, thiserror::Error)]
-pub(crate) enum Error {
-    #[error(transparent)]
-    Client(#[from] influxdb3_client::Error),
-
-    #[error("invlid UTF8 received from server: {0}")]
-    Utf8(#[from] Utf8Error),
-
-    #[error("io error: {0}")]
-    Io(#[from] io::Error),
-}
-
-pub(crate) type Result<T> = std::result::Result<T, Error>;
+use super::common::{create_client, InfluxDb3Config};
 
 #[derive(Debug, Parser)]
 #[clap(visible_alias = "q", trailing_var_arg = true)]
 pub(crate) struct Config {
     /// Common InfluxDB 3.0 config
     #[clap(flatten)]
     influxdb3_config: InfluxDb3Config,
+
+    /// Sampling interval for the queriers. They will perform queries at this interval and
+    /// sleep for the remainder of the interval. Queriers stagger queries by this interval divided
+    /// by the number of queriers.
+    #[clap(
+        short = 'I',
+        long = "query-interval",
+        env = "INFLUXDB3_LOAD_QUERY_SAMPLING_INTERVAL",
+        default_value = "1s"
+    )]
+    sampling_interval: humantime::Duration,
+
+    /// Number of simultaneous queriers. Each querier will perform queries at the specified `interval`.
+    #[clap(
+        short = 'Q',
+        long = "querier-count",
+        env = "INFLUXDB3_LOAD_QUERIER_COUNT",
+        default_value = "1"
+    )]
+    querier_count: usize,
+
+    #[clap(
+        short = 'F',
+        long = "query-format",
+        env = "INFLUXDB3_LOAD_QUERY_FORMAT",
+        value_enum,
+        default_value = "json"
+    )]
+    query_response_format: Format,
+
+    /// The file that will be used to write the results of the run. If not specified, results
+    /// will be written to <spec_name>_results.csv in the current directory.
```

Comment:
Is this related to https://github.com/influxdata/influxdb/issues/24830? By `version_name`, is that something the user enters, or is it the version of the `influxdb3` binary?

I do find it annoying that every time I run the load test, I have to remove the old file, so I was thinking of putting a timestamp in the filename by default as well.

Reply:
Ah yes, that issue is the one. It could just be the influxdb3 binary version (the commit sha) or I was thinking a short name picked by the user. So if they don't specify the name, grab the commit sha as the one.

Having the timestamp in the name of the file at the end is also probably useful. Like YYYY-MM-DD-HH-MM.

Ultimately, I'm imagining we'll want to be able to just specify some dir for where the reports land, and then all files would be written in there. I figure each comparison run will have these files for each version:

- write results
- query results
- cpu
- mem
- disk
- network

So 12 files for any specific spec that we'd like to compare two different build versions. Letting the user just specify the results directory and then automatically naming all the files for them will be useful. That way the analysis app will also be able to build in assumptions about the file names for selectors to show comparisons.

Reply:
I am aligned on all of that and I like the directory approach. If it's okay, I could move these details into #24830 and implement them as part of that issue.

I also think it might be a worth adding a `--version` command to the `influxdb3` binary ([clap supports this](https://docs.rs/clap/latest/clap/macro.crate_version.html)), and potentially an HTTP API that reports it as well  (I believe the old `/ping` API reported the version in v1?).

Reply:
Yes to both of those things 😄 

Let's get this thing merged in so I can start using it for now!